repos/system_upgrade/common/actors/checkfstabmountorder/libraries/checkfstabmountorder.py

```diff
@@ -0,0 +1,102 @@
+from leapp import reporting
+from leapp.libraries.stdlib import api
+from leapp.models import StorageInfo
+
+
+def _get_common_path(path1, path2):
+    """
+    Return the longest common absolute sub-path for pair of given absolute paths.
+
+    Note that this function implements similar functionality as os.path.commonpath(), however this function is not
+    available in python2.7, thus can't be used here.
+    """
+
+    if path1 == '' or path2 == '':
+        return ''
+
+    path1 = path1.strip('/').split('/')
+    path2 = path2.strip('/').split('/')
+
+    common_path = []
+    for path1_part, path2_part in zip(path1, path2):
+        if path1_part != path2_part:
+            break
+        common_path.append(path1_part)
+    return '/' + '/'.join(common_path)
+
+
+def _get_overshadowing_mount_points(mount_points):
+    """
+    Retrieve set of overshadowing and overshadowed mount points.
+
+    :param list[str] mount_points: absolute paths to mount points
+    :returns: set of mount points
+    """
+
+    overshadowing = set()
+    for i, mount_point in enumerate(mount_points):
+        if mount_point[-1] == '/':
+            mount_point = mount_point[:-1]
+
+        for overshadowing_mount_point in mount_points[i+1:]:
+            if overshadowing_mount_point[-1] == '/':
+                overshadowing_mount_point = overshadowing_mount_point[:-1]
+
+            if _get_common_path(mount_point, overshadowing_mount_point) == overshadowing_mount_point:
+                overshadowing.add(overshadowing_mount_point)
+                overshadowing.add(mount_point)
+
+    return overshadowing
+
+
+def check_fstab_mount_order():
+    storage_info = next(api.consume(StorageInfo), None)
+
+    if storage_info:
+
+        mount_points = []
+        for fstab_entry in storage_info.fstab:
+            mount_point = fstab_entry.fs_file
+            if mount_point[-1] == '/':
```

Comment:
imho `rstrip(os.path.sep)` is more readable and correct

IIUC L57-62 can be changed to

`mount_points = [fstab_entry.fs_file.rstrip(os.path.sep) for fstab_entry in storage_info.fstab]`

Reply:
I have a different question:
* Why so many `if ... == '/'` checks?

I would expect to deal with the trailing slash just on one place and never need to think about it again in the code. So we see just one `rstrip('/')` in the code and never need to hear about check/rstrip anywhere else during the execution.

Reply:
Yes, you are right, it could be done using single ``rstrip('/')``. The ``rstrip('/')`` in ``check_fstab_mount_order()`` filters out the trailing ``/`` and we can forget about it, however...

Looking at the function ``_get_overshadowing_mount_points()`` as a standalone sub-program, it should strip the trailing ``/`` either way. Given that it relies on unified format in the comparison (``_get_common_path`` always returns path without the last ``/``) and also when creating its output (set of unique mount points that are involved in overshadowing). Therefore I decided to enforce it inside the function as well, to ensure correct results even without removal of the trailing ``/`` from its input list before calling the function.

This doesn't change the fact that the previous implementation was odd though. I adjusted it and I think that it doesn't feel that off now.